examples/fluids/src/setuplibceed.c

```diff
@@ -313,6 +315,159 @@ PetscErrorCode CreateOperatorForDomain(Ceed ceed, DM dm, SimpleBC bc,
   PetscFunctionReturn(0);
 }
 
+PetscErrorCode DMPlexInsertBoundaryValues_StrongBCCeed(DM dm,
+    PetscBool insert_essential, Vec Q_loc, PetscReal time, Vec face_geom_FVM,
+    Vec cell_geom_FVM, Vec grad_FVM) {
+
+  User user;
+  PetscScalar *q;
+  PetscMemType q_mem_type;
+  PetscFunctionBeginUser;
+
+  PetscCall(DMGetApplicationContext(dm, &user));
+  // Setup libCEED vector
+  PetscCall(VecGetArrayAndMemType(Q_loc, &q, &q_mem_type));
+  CeedVectorSetArray(user->q_ceed, MemTypeP2C(q_mem_type), CEED_USE_POINTER, q);
+
+  // Apply libCEED operator
+  CeedOperatorApply(user->op_dirichlet, CEED_VECTOR_NONE, user->q_ceed,
```

Comment:
Side note - this really should use ApplyAdd

We found a bug with viz by overwriting the target vector.
See one potential fix here:
https://gitlab.com/micromorph/ratel/-/blob/main/src/ratel-petsc-ops.c#L279

Reply:
Right. In Ratel we are manually applying the mask before adding in to fix the correctness of visualization.

Reply:
In the Ratel example, what is stored in `ratel->X[0]`? Not sure how best to build the mask in this context.

Reply:
X[0] is the global solution vector. To create the mask, I set the local vector to 0, set the global vector to 1, and then did a global to local.